CRM/Core/I18n/Schema.php

```diff
@@ -44,56 +44,26 @@ public static function dropAllViews() {
    *   the first locale to create (migrate to).
    */
   public static function makeMultilingual($locale) {
+    $isUpdateDone = FALSE;
     $domain = new CRM_Core_DAO_Domain();
-    $domain->find(TRUE);
-
-    // break early if the db is already multi-lang
-    if ($domain->locales) {
-      return;
-    }
-
-    $dao = new CRM_Core_DAO();
-
-    // build the column-adding SQL queries
-    $columns = CRM_Core_I18n_SchemaStructure::columns();
-    $indices = CRM_Core_I18n_SchemaStructure::indices();
-    $queries = [];
-    foreach ($columns as $table => $hash) {
-      // drop old indices
-      if (isset($indices[$table])) {
-        foreach ($indices[$table] as $index) {
-          if (CRM_Core_BAO_SchemaHandler::checkIfIndexExists($table, $index['name'])) {
-            $queries[] = "DROP INDEX {$index['name']} ON {$table}";
-          }
-        }
-      }
-      // deal with columns
-      foreach ($hash as $column => $type) {
-        $queries[] = "ALTER TABLE {$table} ADD {$column}_{$locale} {$type}";
-        if (CRM_Core_BAO_SchemaHandler::checkIfFieldExists($table, $column)) {
-          $queries[] = "UPDATE {$table} SET {$column}_{$locale} = {$column}";
-          $queries[] = "ALTER TABLE {$table} DROP {$column}";
-        }
+    $domain->find();
+    while ($domain->fetch()) {
+      // skip if the domain is already multi-lang.
+      if ($domain->locales) {
+        continue;
```

Comment:
Should we make sure that all domains have the same locales?

Reply:
@seamuslee001 I dunno - I think one domain could have less than the others but it gets tricky fast. I decided not to change that edge case

Reply:
@colemanw was thinking that instead maybe domain1 should be treated as  global domain see other discussion here https://github.com/civicrm/civicrm-core/pull/17650